### PR TITLE
Set Required Coverage to 100

### PR DESCRIPTION
**Pull request by ghickman**
This raises the minimum coverage to 100% and fixes each of the remaining failures/warnings in the coverage report.  Each commit explains why a failure/warning was tackled in the way it was to make it to track them in history should we need to.

### CODE REVIEW

File: databuilder/query_engines/base_sql.py [hunk 1]
```diff
@@ -273,6 +273,17 @@ def get_sql_element_no_cache(self, node: QueryNode) -> ClauseElement:
...
+        """
+        Certain places in our Query Model support values which can either be QueryNodes
+        themselves (which require resolving to SQL) or plain static values (booleans,
+        integers, dates, list of dates etc) which we can return unchanged to SQLAlchemy.
+        """
+        if isinstance(value, QueryNode):
+            return self.get_sql_element(value)
+        else:
+            return value
+
```
Comment (evansd) [inline databuilder/query_engines/base_sql.py:286]: I think a rebase has gone wrong here, and this commit should actually now be a no-op.

File: databuilder/query_engines/base_sql.py [hunk 2]
```diff
@@ -238,9 +238,9 @@ def get_static_value(self, value: Union[Scalar, tuple, list]) -> StaticValue:
         # their elements at runtime. See: https://bugs.python.org/issue46191
         if isinstance(value, (tuple, list)) and any(
             not isinstance(v, SCALAR_TYPES) for v in value
-        ):
+        ):  # pragma: no cover
             assert False, f"Unhandled type {value!r}"
-        return value
+        return value  # pragma: no cover
```
Comment (evansd) [inline databuilder/query_engines/base_sql.py:243]: Just for my own education, how come this line needs `no cover`? Isn't the return always executed here

**Review by evansd — APPROVED** [review]
Thanks for doing this George, I think it's really going to pay off as development continues.

I think there's just one commit that needs removing (ae2ea897b5d90c9b0fde89a00abbf297311e1baa) which I'm pretty sure is just git rebase not following the fact that a function was moved to a different place (in this commit 80de1ce30c6bac51f2251aa8475aa06ae2e080c2).

File: databuilder/query_engines/base_sql.py (hunk 2 above)
Comment (ghickman) [reply databuilder/query_engines/base_sql.py]: I added it without thinking 🤦‍♂️  Fixed and rebased now.

**Pull request merged**